main.js

```diff
@@ -1,163 +1,370 @@
-    type: '32WL1A63DG',
-    name: 'HD TV',
-    brand: 'Toshiba',
-    price: 161,
-    availableSizes: [32],
-    refreshRate: 50,
-    screenType: 'LED',
-    screenQuality: 'Full HD',
-    smartTv: false,
-    options: {
-      wifi: false,
-      speech: false,
-      hdr: true,
-      bluetooth: false,
-      ambiLight: false,
+    {
+        type: '32WL1A63DG',
+        name: 'HD TV',
+        brand: 'Toshiba',
+        price: 161,
+        availableSizes: [32],
+        refreshRate: 50,
+        screenType: 'LED',
+        screenQuality: 'Full HD',
+        smartTv: false,
+        options: {
+            wifi: false,
+            speech: false,
+            hdr: true,
+            bluetooth: false,
+            ambiLight: false,
+        },
+        originalStock: 10,
+        sold: 8,
     },
-    originalStock: 10,
-    sold: 8,
-  },
-];
\ No newline at end of file
+];
+
+//Opdracht 1a: Gebruik een array-methode om een array te maken met alle tv-type namen. Log de uitkomst in de console.
+
+const tvType = inventory.map((item) => {
+    return item.type;
+});
+
+console.log("tvType:" + tvType);
+
+// Opdracht 1b: Gebruik een array-methode om alle tv's te verzamelen (de hele objecten) die volledig uitverkocht zijn. Log de uitkomst in de console.
+
+function soldout() {
+    const soldOut = inventory.filter((item) => {
+        let calculateSoldOut = item.originalStock - item.sold;
+        return calculateSoldOut <= 0;
+    });
+    console.log(soldOut);
+}
+
+
+// Opdracht 1c: Gebruik een array-methode om alle tv's te verzamelen (de hele objecten) die over AmbiLight beschikken. Log de uitkomst in de console.
+
+function ambilight() {
+    const ambiLight = inventory.filter((item) => {
+        return item.options.ambiLight;
+    });
+    console.log(ambiLight);
+}
+
+// Opdracht 1d: Schrijf een functie die alle tv's van laagste naar hoogste prijs sorteert. Log de uitkomst in de console.
+
+function sortPrice() {
+    const sortPrice = inventory.sort((a, b) => {
+        return a.price - b.price;
+    });
+    console.log(sortPrice);
+}
+
+// Opdracht 2a: Hoeveel tv's zijn er al verkocht? Schrijf een script dat dit berekent. Log de uitkomst in de console.
+
+let countSoldTv = 0;
+
+for (let i = 0; i < inventory.length; i++) {
+    countSoldTv += inventory[i].sold;
+}
+
+console.log(countSoldTv);
+
+//Opdracht 2b: Zorg ervoor dat dit aantal in het groen wordt weergegeven op de pagina.
+
+const elementSoldTV = document.getElementById("sold-tv")
+elementSoldTV.textContent = countSoldTv;
+
+// Opdracht 2c: Hoeveel tv's heeft Tech It Easy ingekocht? Schrijf een script dat dit berekent. Log de uitkomst in de console.
+
+let countPurchasedTv = 0;
+
+for (let i = 0; i < inventory.length; i++) {
+    countPurchasedTv += inventory[i].originalStock;
+}
+
+console.log(countPurchasedTv);
+
+// Opdracht 2d: Zorg ervoor dat dit aantal in het blauw wordt weergegeven op de pagina.
+
+const elementPurchasedTV = document.getElementById("purchashed-tv")
```

Comment:
Ik zie dat je hier en bij 235 achter een const declaratie niet eindigt met een ;. Is dat bewust?

Reply:
Goed gezien, hier zou ook een ; mogen. ik zal de code updaten.